### PR TITLE
 RenderStatePackager: Minor changes and updates

### PR DESCRIPTION
**Pull request by MikhailGorobets**

### CODE REVIEW

File: RenderStatePackager/src/RenderStatePackager.cpp [hunk 1]
```diff
@@ -36,35 +39,59 @@
...
-            return m_CurrentPath + FileSystem::GetSlashSymbol() + FileName;
+            return CombinePath(FileName);
         }
 
     private:
-        String m_CurrentPath;
+        String CombinePath(const Char* Path) const
+        {
+            std::stringstream  Stream;
+            std::deque<String> Queue(m_Paths);
```
Comment (TheMostDiligent) [inline RenderStatePackager/src/RenderStatePackager.cpp:84]: No need to copy the deque!
You can iterate it like any other container: `for (const auto& Path : m_Paths)`

File: RenderStatePackager/src/RenderStatePackager.cpp [hunk 2]
```diff
@@ -24,6 +24,9 @@
  *  of the possibility of such damages.
  */
 
+#include <deque>
+#include <sstream>
+
 #include "RenderStatePackager.hpp"
```
Comment (TheMostDiligent) [inline RenderStatePackager/src/RenderStatePackager.cpp:30]: Please have have the main header (`#include "RenderStatePackager.hpp"`) the first header in the file
https://github.com/DiligentGraphics/DiligentCore/blob/master/doc/CodingGuidelines.md#source-files

File: Tests/DiligentToolsTest/assets/RenderStates/RenderStatePackager/GraphicsPrimitivesDump.json [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+            "EntryPoint": "PSBlitTexture",
+            "UseCombinedTextureSamplers": true
+        },
+        {
+            "Desc": {
+                "Name": "ClearUnorderedAccessViewUint CS",
+                "ShaderType": "COMPUTE"
+            },
+            "SourceLanguage": "HLSL",
+            "FilePath": "GraphicsPrimitives.hlsl",
```
Comment (TheMostDiligent) [inline Tests/DiligentToolsTest/assets/RenderStates/RenderStatePackager/GraphicsPrimitivesDump.json:29]: It is better to use some descriptive name, e.g. BrokenShader.hlsl and don't have any actual code in it (ClearUnorderedAccessViewUint does not add anything), e.g.

BrokenShader.hlsl
`Broken shader code;`

Also, why do you need multiple pipelines in the JSON. It looks like single pipeline will be enough.

File: Tests/DiligentToolsTest/src/RenderStatePackager/RenderStatePackagerTest.cpp [hunk 4]
```diff
@@ -248,4 +251,72 @@ TEST(Tools_RenderStatePackager, PackagerMissingObjectsTest)
...
+            return ".dxbc";
+        case ARCHIVE_DEVICE_DATA_FLAG_VULKAN:
+            return ".spv";
+        case ARCHIVE_DEVICE_DATA_FLAG_METAL_IOS:
+        case ARCHIVE_DEVICE_DATA_FLAG_METAL_MACOS:
+            return ".air";
+        case ARCHIVE_DEVICE_DATA_FLAG_GL:
+        case ARCHIVE_DEVICE_DATA_FLAG_GLES:
+            return ".hlsl";
+        default:
```
Comment (TheMostDiligent) [inline Tests/DiligentToolsTest/src/RenderStatePackager/RenderStatePackagerTest.cpp:269]: On MacOS please also verify that .metal and .metallib files are created as well

**Review by TheMostDiligent — CHANGES_REQUESTED** [review]

File: Tests/DiligentToolsTest/assets/RenderStates/RenderStatePackager/GraphicsPrimitivesDump.json (hunk 3 above)
Comment (MikhailGorobets) [reply Tests/DiligentToolsTest/assets/RenderStates/RenderStatePackager/GraphicsPrimitivesDump.json:29]: We need several pipeline descriptions to test the dump with different types of pipelines, and also to make sure that the dump works for all pipelines of the same type.

**Comment by TheMostDiligent** [comment]
Merged in https://github.com/DiligentGraphics/DiligentTools/commit/6e28577419c81a662dca25f6bee8e5849deb8a0e

**Pull request closed without merging**